apps/api/src/campaign/campaign.service.ts

```diff
@@ -57,17 +57,29 @@ export class CampaignService {
   }
 
   async listAllCampaigns(): Promise<Campaign[]> {
-    return await this.prisma.campaign.findMany({
+    const campaigns = await this.prisma.campaign.findMany({
       orderBy: {
         endDate: 'asc',
       },
       include: {
         campaignType: { select: { name: true } },
         beneficiary: { select: { person: { select: { firstName: true, lastName: true } } } },
         coordinator: { select: { person: { select: { firstName: true, lastName: true } } } },
-        vaults: { select: { id: true, amount: true } },
+        vaults: { select: { donations: { select: { amount: true } }, amount: true } },
+        incomingTransfers: { select: { amount: true } },
+        outgoingTransfers: { select: { amount: true } },
       },
     })
+    for (const campaign of campaigns) {
```

Comment:
@borislavstoychev Have you checked if you can reuse `addReachedAmountAndDonors` method?

Reply:
I agree with Ilko here - please use the calculated totals on server side, and no need to return all separate amounts to frontend